src/ragger/utils/bip/seed.py

```diff
@@ -0,0 +1,44 @@
+from typing import Tuple
+from enum import Enum, auto
+
+from mnemonic import Mnemonic
+from bip_utils import Bip32Secp256k1
+from bip_utils import Bip32Nist256p1
+from bip_utils import Bip32Ed25519Slip
+from bip_utils import Bip32Ed25519Kholaw
+from bip_utils import Bip32Ed25519Blake2bSlip
+
+
+class CurveChoice(Enum):
+    Secp256k1 = auto()
+    Nist256p1 = auto()
+    Ed25519Slip = auto()
+    Ed25519Kholaw = auto()
+    Ed25519Blake2bSlip = auto()
+
+
+GET_CURVE_OBJ = {
+    CurveChoice.Secp256k1: Bip32Secp256k1,
+    CurveChoice.Nist256p1: Bip32Nist256p1,
+    CurveChoice.Ed25519Slip: Bip32Ed25519Slip,
+    CurveChoice.Ed25519Kholaw: Bip32Ed25519Kholaw,
+    CurveChoice.Ed25519Blake2bSlip: Bip32Ed25519Blake2bSlip,
+}
+
+SPECULOS_MNEMONIC = (
+    "glory promote mansion idle axis finger extra february uncover one trip resource "
+    "lawn turtle enact monster seven myth punch hobby comfort wild raise skin")
+
+
+def calculate_public_key_and_chain_code(curve: CurveChoice,
```

Comment:
Maybe you can add a test for this function?

Reply:
Done